auth/mux.go

```diff
@@ -0,0 +1,119 @@
+package auth
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+)
+
+/*
+   When establishing a connection to the mux, in addition to the address of the receiver,
+   the sender sends an authentication token and signs the whole message. The token determines
+   if the sender is authorized to send data to the receiver or not
+
+   ----------------------------------------------------------------------------------------
+   | address (6 bytes) |  token length (4 bytes)  |   Auth Token (N bytes)  |  Signature  |
```

Comment:
Maybe totally academic, but I'd put the address after the token, so you don't read those extra 6 bytes if the token is invalid.